js/render/draw_collision_debug.js

```diff
@@ -2,33 +2,39 @@
 
 module.exports = drawCollisionDebug;
 
-function drawCollisionDebug(painter, layer, coord, tile) {
-    if (!tile.buffers) return;
-    var elementGroups = tile.getElementGroups(layer, 'collisionBox');
-    if (!elementGroups) return;
-
+function drawCollisionDebug(painter, source, layer, coords) {
     var gl = painter.gl;
-    var buffer = tile.buffers.collisionBoxVertex;
     var shader = painter.collisionBoxShader;
-    var posMatrix = painter.calculatePosMatrix(coord);
-
     gl.enable(gl.STENCIL_TEST);
-    painter.enableTileClippingMask(coord);
+    gl.switchShader(shader);
+
+    for (var i = 0; i < coords.length; i++) {
+        var coord = coords[i];
+        var tile = source.getTile(coord);
+        var elementGroups = tile.getElementGroups(layer, 'collisionBox');
+        if (!elementGroups || !tile.buffers) continue;
+
+        var buffer = tile.buffers.collisionBoxVertex;
+        buffer.bind(gl);
+        buffer.setAttribPointers(gl, shader, 0);
 
-    gl.switchShader(shader, posMatrix);
+        var posMatrix = painter.calculatePosMatrix(coord, source.maxzoom);
+        gl.setPosMatrix(posMatrix);
 
-    buffer.bind(gl);
-    buffer.setAttribPointers(gl, shader, 0);
+        painter.enableTileClippingMask(coord);
 
-    gl.lineWidth(1);
+        gl.lineWidth(1);
+        gl.uniform1f(shader.u_scale, Math.pow(2, painter.transform.zoom - tile.coord.z));
+        gl.uniform1f(shader.u_zoom, painter.transform.zoom * 10);
+        gl.uniform1f(shader.u_maxzoom, (tile.coord.z + 1) * 10);
 
-    gl.uniform1f(shader.u_scale, Math.pow(2, painter.transform.zoom - tile.coord.z));
-    gl.uniform1f(shader.u_zoom, painter.transform.zoom * 10);
-    gl.uniform1f(shader.u_maxzoom, (tile.coord.z + 1) * 10);
+        gl.drawArrays(
+            gl.LINES,
+            elementGroups.groups[0].vertexStartIndex,
```

Comment:
this line already existed before this pr, but does `.groups[0]` always exist at this point?